### PR TITLE
STMACOM-483 Allow the selection of remote storage locations

### PR DESCRIPTION
**Pull request by uladislausamets**
## Purpose

https://issues.folio.org/browse/STSMACOM-483
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: lib/LocationSelection/LocationInput.js [hunk 1]
```diff
@@ -50,12 +53,21 @@ export default class LocationInput extends React.Component {
...
+    const { resources, placeholder, remoteMap, intl } = this.props;
     const locations = (resources.locations || {}).records || [];
 
     if (!locations.length) return (<div />);
 
-    const locationOpts = locations.map(loc => ({ label: `${loc.name} (${loc.code})`, value: loc.id }));
+    const locationOpts = locations.map(loc => {
+      const remoteLocationLabel = (
+        remoteMap && (loc.id in remoteMap)
+      ) ? intl.formatMessage({ id: 'stripes-smart-components.remoteLabel' }) : '';
```
Comment (aliaksei-chumakou) [inline lib/LocationSelection/LocationInput.js:64]: IMHO, it's better to move it above and make one time call, use result after
`const remoteLabel = intl.formatMessage({ id: 'stripes-smart-components.remoteLabel' });`

File: translations/stripes-smart-components/en.json [hunk 2]
```diff
@@ -59,6 +59,7 @@
   "ll.permanent": "permanent",
   "ll.saveAndClose": "Save and close",
   "ll.cancel": "Cancel",
+  "ll.remoteLabel": "(Remote)",
```
Comment (aliaksei-chumakou) [inline translations/stripes-smart-components/en.json:62]: if you look at the `(${loc.code})` part, parenthesis leveraged to code there. Maybe it makes sense to be consistent with this suffix as well?

File: lib/RemoteStorageTag/RemoteStorageTag.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import { useRemoteStorageMappings } from '../utils';
+
+const RemoteStorageTag = ({ location }) => {
+  const remoteMap = useRemoteStorageMappings();
+  const intl = useIntl();
+  const remoteLabel = `(${intl.formatMessage({ id: 'stripes-smart-components.remoteLabel' })})`;
+
+  return (
+    remoteMap && (location.id in remoteMap)
+  ) ? remoteLabel : '';
```
Comment (axelhunn) [inline lib/RemoteStorageTag/RemoteStorageTag.js:13]:
```suggestion

  return (location.id in remoteMap) && `(${intl.formatMessage({ id: 'stripes-smart-components.remoteLabel' })})`;
```

**Review by axelhunn — APPROVED** [review]

File: lib/utils/index.js [hunk 4]
```diff
@@ -0,0 +1,2 @@
+// eslint-disable-next-line import/prefer-default-export
+export { default as useRemoteStorageMappings } from './useRemoteStorageMappings';
```
Comment (axelhunn) [inline lib/utils/index.js:2]:
```suggestion
export { default as useRemoteStorageMappings } from './useRemoteStorageMappings';
export default {};
```

**Review by aliaksei-chumakou — APPROVED** [review]

File: index.js [hunk 5]
```diff
@@ -69,3 +69,5 @@ export { default as ViewCustomFieldsSettings } from './lib/CustomFields/pages/Vi
 export { default as EditCustomFieldsSettings } from './lib/CustomFields/pages/EditCustomFieldsSettings';
 export { default as EditCustomFieldsRecord } from './lib/CustomFields/pages/EditCustomFieldsRecord';
 export { default as ViewCustomFieldsRecord } from './lib/CustomFields/pages/ViewCustomFieldsRecord';
+
+export * from './lib/utils';
```
Comment (aliaksei-chumakou) [inline index.js:73]: IMHO it's better to keep code style consistent - here it's hard to say what you are exporting actually...

Comment (axelhunn) [reply index.js:73]: I proposed this change in fact.
For now it's like "just re-export whatever it is there".
Later we'll maybe have a whole bunch of API-related hooks using react-query, then we'll figure out the proper directory/module structure.

**Pull request merged**